src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -19,9 +19,33 @@ internal static class DeserializationHelpers
     {
         internal static void ApplyProperty(ODataProperty property, IEdmStructuredTypeReference resourceType, object resource,
             ODataDeserializerProvider deserializerProvider, ODataDeserializerContext readContext)
-        { 
-            IEdmProperty edmProperty = resourceType.FindProperty(property.Name);
-                       
+        {
+            IEdmProperty edmProperty;
+
+            // Extract edmProperty using a Case-Sensitive match
+            edmProperty = resourceType.FindProperty(property.Name);
+
+            if (readContext != null && !readContext.DisableCaseInsensitiveRequestPropertyBinding && edmProperty == null)
+            {
+                // if edmProperty is null, we try a Case-Insensitive match.
+                bool propertyMatched = false;
+
+                IEnumerable<IEdmStructuralProperty> sProperties = resourceType.StructuralProperties();
+                foreach(IEdmStructuralProperty sProperty in sProperties)
```

Comment:
```suggestion
                foreach(IEdmStructuralProperty structuralProperty in structuralProperties)
```
I'd vote for the longer unambiguous names

Reply:
#Resolved